File: Tema 3 - Web API/PetShelter/PetShelter.DataAccessLayer/Models/Fundraiser.cs
using System;
using Microsoft.VisualBasic;

namespace PetShelter.DataAccessLayer.Models
{
    public class Fundraiser : IEntity
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public decimal CurrentlyDonationAmount { get; set; }

        public decimal GoalValue { get; set; }

        public DateTime CreationDate { get; set; }

        public DateTime DueDate { get; set; }

        public ICollection<Donation> Donations { get; set; }

        public Person Owner { get; set; }

        public string Status { get; set; }

        public int OwnerId { get; set; }

        public Fundraiser(int id, string name, decimal goalValue, DateTime dueDate)
        {
            Name = name;
            GoalValue = goalValue;
            CurrentlyDonationAmount = 0;
            CreationDate = DateTime.Now;
            DueDate = dueDate;
            Donations = new List<Donation>();
            Status = "Active";
            Owner = new Person();
        }
    }
}

File: Tema 5 - Typescript/after/PetShelterFrontend/src/models/Fundraiser.ts
import type { Person } from './Person';

export class Fundraiser {
  name: string;
  goalValue: number;
  currentlyRaisedAmount?: number;
  dueDate: Date;
  creationDate?: Date;
  status?: Status;
  owner: Person;

  constructor(
    name: string,
    goalValue: number,
    dueDate: Date,
    owner: Person,
    creationDate?: Date,
    status?: Status,
    currentlyRaisedAmount?: number,
  ) 
  
  {
    this.name = name;
    this.goalValue = goalValue;
    this.currentlyRaisedAmount = currentlyRaisedAmount;
    this.creationDate = creationDate;
    this.dueDate = dueDate;
    this.status = status;
    this.owner = owner;
  }
}

export const enum Status {
  'Closed' = 0,
  'Active' = 1,
}
